include/rocksdb/status.h

```diff
@@ -73,6 +73,7 @@ class Status {
     kStaleFile = 6,
     kMemoryLimit = 7,
     kSpaceLimit = 8,
+    kPathNotFound = 9,
```

Comment:
Don't we already use kNotFound for this specific case? So, why do we need a new one?

Reply:
The `Code::kNotFound` is intended to denote key-value pair is not found. `File not found` is an `IOError` and should be represented with the combination of `IOError` and a proper subcode. Let me know if I have missed any background context on the meaning of `kNotFound`.

Reply:
Chatted offline with @siying . `Open`ing a non-existing file for read is an `IOError`.
`TableCache` should return `IOError` instead of `NotFound` when the file does not exist. Therefore, we still need a new subcode.